src/storage/src/hummock/sstable_store.rs

```diff
@@ -28,7 +28,26 @@ const MAX_META_CACHE_SHARD_BITS: usize = 2;
 const MAX_CACHE_SHARD_BITS: usize = 6; // It means that there will be 64 shards lru-cache to avoid lock conflict.
 const MIN_BUFFER_SIZE_PER_SHARD: usize = 256 * 1024 * 1024; // 256MB
 
-pub type TableHolder = CachableEntry<HummockSstableId, Box<Sstable>>;
+pub enum TableHolder {
+    CachedMeta(CachableEntry<HummockSstableId, Box<Sstable>>), /* This should never hold block
+                                                                * data. */
+    Owned(Box<Sstable>),
+}
+
+impl TableHolder {
+    pub fn value(&self) -> &Sstable {
+        match self {
+            Self::CachedMeta(meta_only_sst) => meta_only_sst.value(),
+            Self::Owned(sst) => sst,
+        }
+    }
+}
+
+impl From<CachableEntry<HummockSstableId, Box<Sstable>>> for TableHolder {
+    fn from(cached_meta: CachableEntry<HummockSstableId, Box<Sstable>>) -> Self {
```

Comment:
Not sure if this would copy the entry cachableentry...? But the underlying data are just pointers so its cheap.